### ISSUE TITLE
Improvement of the build system

### ISSUE THREAD

**Issue by daisukenishino** [OpenTouryoProject/OpenTouryo#119]
# Improvement item
## BAT
- Correct the *.bat file.
- Target  
  - https://github.com/OpenTouryoProject/OpenTouryo/tree/develop/root/programs/C%23
  - https://github.com/OpenTouryoProject/OpenTouryo/tree/develop/root/programs/VB

#### Issue
The path to the devenv.exe and msbuild.exe are different.
- 64bit program :  
  C:\Program Files
- 32bit program :  
  C:\Program Files (x86)

#### Solution
I want to change the automatically path by determining the x86 or x64 in the  bat.

#### Issue
Build.bat file of PortableClassLibrary is not exist.

#### Solution
Add the Build.bat file of PortableClassLibrary to appropriate location.

### MSbuild
Please replace the contents of the following files.
MSBuild will be used as default tool by this changes.

##### Target
- https://github.com/OpenTouryoProject/OpenTouryo/blob/develop/root/programs/C%23/
  - z_Common.bat
  - z_Common2.bat
- https://github.com/OpenTouryoProject/OpenTouryo/blob/develop/root/programs/VB/
  - z_Common.bat
  - z_Common2.bat

And, please confirm that the build will pass.

### NuGet (Only VS2015)
Begin correspondence to NuGet from VS2015.

##### Target  
https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015

Following libraries and associated libraries exist on the Nuget.

##### Components
- [Public project](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/Frameworks/Infrastructure/Public/Public.csproj) is the target.
  - log4net (log4net.dll)
  - dotnetzip (Ionic.Zip.dll)
  - json.net (Newtonsoft.Json.dll)
- [Framework project](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/Frameworks/Infrastructure/Framework/Framework.csproj) is the target.
  - System.Web.Mvc.dll
- [Business project](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/Frameworks/Tools/DeployZipPackWithHTTP/DeployZipPackWithHTTP.csproj) is the target.
  - System.Web.Mvc.dll
- [DeployZipPackWithHTTP project]()
  - dotnetzip (Ionic.Zip.dll)

##### Data providers
- Each Dam project is the target.
  - ODP.NET (Oracle.DataAccess.dll)
  - Npgsql (Npgsql.dll)
  - MySQL Connector/NET (MySql.Data.dll)

But, [DamDB2](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/Frameworks/Infrastructure/Public/Db/DamDB2/DamDB2.csproj) and [DamHiRDB](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/Frameworks/Infrastructure/Public/Db/DamHiRDB/DamHiRDB.csproj) are not target because appropriate Nuget library does not exist.

##### Samples
- [MVC_Sample] is the target.(https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015/programs/C%23/Samples/WebApp_sample/MVC_Sample)
- [SPA_Sample] is the target.(https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015/programs/C%23/Samples/WebApp_sample/SPA_Sample)
  - System.Web.Mvc.dll

##### Remove and restore packages folder
Finally, I want to remove the packages folder. And, the contents of the packages folder want to be acquired by the NuGet. Therefor, do not remove package.config file from the projects or uninstall nuget packages because when you build the project, the required packages will be downloaded from the NuGet.

And, Set to the "Enable NuGet Package Restore" property on Visual Studio project, ".nuget" folder (containing NuGet.exe and NuGet.targets files) will be created in solution folder. And, MSBuild will call NuGet.exe, and NuGet.exe will restore NuGet packages when perform build task using this settings. Therefore, this correspondance will not affect current build system.

see : https://docs.nuget.org/consume/package-restore/msbuild-integrated

 followings are target of this correspondance.

- Frameworks
  - AllComponent.sln
    - Business.csproj
    - Framework.csproj
    - Public.csproj
  - Dam
    - /Public/Db/AllDam.sln (create new. then, notice at the path.)
      - DamMySQL.csproj
      - DamOraOdp.csproj
      - DamPstGrS.csproj
  - Tools
     - DeployZipPackWithHTTP.csproj (.sln)
- Samples
  - MVC_Sample.csproj (.sln)
  - SPA_Sample.csproj (.sln)

**Comment by daisukenishino** [comment]
@SandeepNayak077 

#### About Correspondence to NuGet 
- NuGet correspondence is done to *.csproj or *.vbproj file of later version of VS2015.
- If all of the DLL file reference will be Nuget, [./Public/Dll](https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015/programs/C%23/Frameworks/Infrastructure/Public/Dll) folder will be no longer required.
- I think that it is good to place the packages folder to the location of the [./C#/Frameworks](
https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015/programs/C%23/Frameworks).


#### About build of PortableClassLibrary
- Please named the *.bat file to "3_Build_PortableClassLibrary.bat".
- You can not build a project of the PCL by the VS express edition, but I think that it can be built by changing the MSbuild to the default build tool by replacing the contents of the z_Common.bat and z_Common2.bat.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

Thank you for the information.
We will follow as discribed above.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

This is error log that occurred in the VS2015 Professional installed environment.
I got report that is not failed in case of using the "Developer Command Prompt For VS2015".

Please investigate the measures of above problem "The build using MSBuild failed".

```TXT
\root_VS2015\programs\C#>setlocal
--------------------------------------------------
The choice of build configuration (Debug / Release).
--------------------------------------------------
--------------------------------------------------
Creating a build command.
--------------------------------------------------
サブディレクトリまたはファイル Samples\WebApp_sample\ProjectX_sample\Bin は既に
存在します。
Frameworks\Infrastructure\Build\Business.dll
Frameworks\Infrastructure\Build\Business.pdb
Frameworks\Infrastructure\Build\Business.RichClient.dll
Frameworks\Infrastructure\Build\Business.RichClient.pdb
Frameworks\Infrastructure\Build\Business.XML
Frameworks\Infrastructure\Build\CustomControl.dll
Frameworks\Infrastructure\Build\CustomControl.pdb
Frameworks\Infrastructure\Build\CustomControl.RichClient.dll
Frameworks\Infrastructure\Build\CustomControl.RichClient.pdb
Frameworks\Infrastructure\Build\CustomControl.XML
Frameworks\Infrastructure\Build\Framework.dll
Frameworks\Infrastructure\Build\Framework.pdb
Frameworks\Infrastructure\Build\Framework.RichClient.dll
Frameworks\Infrastructure\Build\Framework.RichClient.pdb
Frameworks\Infrastructure\Build\Framework.xml
Frameworks\Infrastructure\Build\Ionic.Zip.dll
Frameworks\Infrastructure\Build\log4net.dll
Frameworks\Infrastructure\Build\Newtonsoft.Json.dll
Frameworks\Infrastructure\Build\Public.dll
Frameworks\Infrastructure\Build\Public.pdb
Frameworks\Infrastructure\Build\Public.xml
Frameworks\Infrastructure\Build\ja-JP\Business.resources.dll
Frameworks\Infrastructure\Build\ja-JP\Framework.resources.dll
Frameworks\Infrastructure\Build\ja-JP\Public.resources.dll
24 個のファイルをコピーしました
Samples\WS_sample\Build\Business.dll
Samples\WS_sample\Build\Business.pdb
Samples\WS_sample\Build\Business.xml
Samples\WS_sample\Build\Framework.dll
Samples\WS_sample\Build\Framework.pdb
Samples\WS_sample\Build\Framework.xml
Samples\WS_sample\Build\Ionic.Zip.dll
Samples\WS_sample\Build\log4net.dll
Samples\WS_sample\Build\Newtonsoft.Json.dll
Samples\WS_sample\Build\Public.dll
Samples\WS_sample\Build\Public.pdb
Samples\WS_sample\Build\Public.xml
Samples\WS_sample\Build\WSIFType_sample.dll
Samples\WS_sample\Build\WSIFType_sample.pdb
Samples\WS_sample\Build\WSServer_sample.dll
Samples\WS_sample\Build\WSServer_sample.pdb
Samples\WS_sample\Build\ja-JP\Business.resources.dll
Samples\WS_sample\Build\ja-JP\Framework.resources.dll
Samples\WS_sample\Build\ja-JP\Public.resources.dll
19 個のファイルをコピーしました
続行するには何かキーを押してください . . .
Microsoft (R) Build Engine バージョン 4.6.81.0
[Microsoft .NET Framework、バージョン 4.0.30319.42000]
Copyright (C) Microsoft Corporation. All rights reserved.

このソリューション内のプロジェクトを 1 度に 1 つずつビルドします。並行ビルドを有
効にするには、"/m" スイッチを追加してください。
2015/10/30 17:14:44 にビルドを開始しました。
ノード 1 上のプロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C
#\Samples\
WebApp_sample\ProjectX_sample\ProjectX_sample.sln" (既定のターゲット)。
ValidateSolutionConfiguration:
  ソリューション構成 "Debug|Any CPU" をビルドしています。
プロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\Web
App_s
ample\ProjectX_sample\ProjectX_sample.sln" (1) は、ノード 1 上に "C:\OpenTouryoT
emplat
es-nishino\root_VS2015\programs\C#\Samples\WebApp_sample\ProjectX_sample\Projec
tX_sample.metaproj" (2) をビルドしています (既定のターゲット)。
Build:
  C:\Windows\Microsoft.NET\Framework\v4.0.30319\aspnet_compiler.exe -v /Project
  X_sample -p ..\ProjectX_sample\ -u -f -d ..\PrecompiledWeb\ProjectX_sample\
プロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\Web
App_s
ample\ProjectX_sample\ProjectX_sample.metaproj" (既定のターゲット) のビルドが完
了しました。

プロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\Web
App_s
ample\ProjectX_sample\ProjectX_sample.sln" (既定のターゲット) のビルドが完了しま
した。


ビルドに成功しました。
    0 個の警告
    0 エラー

経過時間 00:00:17.46
続行するには何かキーを押してください . . .
Microsoft (R) Build Engine バージョン 4.6.81.0
[Microsoft .NET Framework、バージョン 4.0.30319.42000]
Copyright (C) Microsoft Corporation. All rights reserved.

このソリューション内のプロジェクトを 1 度に 1 つずつビルドします。並行ビルドを有
効にするには、"/m" スイッチを追加してください。
2015/10/30 17:15:05 にビルドを開始しました。
ノード 1 上のプロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C
#\Samples\
WebApp_sample\MVC_Sample\MVC_Sample.sln" (既定のターゲット)。
ValidateSolutionConfiguration:
  ソリューション構成 "Debug|Any CPU" をビルドしています。
プロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\Web
App_s
ample\MVC_Sample\MVC_Sample.sln" (1) は、ノード 1 上に "C:\OpenTouryoTemplates-n
ishino
\root_VS2015\programs\C#\Samples\WebApp_sample\MVC_Sample\MVC_Sample\MVC_Sample
.csproj" (2) をビルドしています (既定のターゲット)。
C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\WebApp_sample\MV
C_Sample\MVC_Sample\MVC_Sample.csproj(288,3): error MSB4019: インポートされたプ
ロジェクト "C:
\Program Files\MSBuild\Microsoft\VisualStudio\v11.0\WebApplications\Microsoft.W
ebApplication.targets" が見つかりませんでした。<Import> 宣言のパスが正しいかどう
か、およびファイルがディスクに存在しているかど
うかを確認してください。
プロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\Web
App_s
ample\MVC_Sample\MVC_Sample\MVC_Sample.csproj" (既定のターゲット) のビルドが終了
しました -- 失敗。

プロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\Web
App_s
ample\MVC_Sample\MVC_Sample.sln" (既定のターゲット) のビルドが終了しました -- 失
敗。


ビルドに失敗しました。

"C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\WebApp_sample\M
VC_Sample\MVC_Sample.sln" (既定のターゲット) (1) ->
"C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\WebApp_sample\M
VC_Sample\MVC_Sample\MVC_Sample.csproj" (既定のターゲット) (2) ->
  C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\WebApp_sample\
MVC_Sample\MVC_Sample\MVC_Sample.csproj(288,3): error MSB4019: インポートされた
プロジェクト "
C:\Program Files\MSBuild\Microsoft\VisualStudio\v11.0\WebApplications\Microsoft
.WebApplication.targets" が見つかりませんでした。<Import> 宣言のパスが正しいかど
うか、およびファイルがディスクに存在している
かどうかを確認してください。

    0 個の警告
    1 エラー

経過時間 00:00:00.79
続行するには何かキーを押してください . . .
Microsoft (R) Build Engine バージョン 4.6.81.0
[Microsoft .NET Framework、バージョン 4.0.30319.42000]
Copyright (C) Microsoft Corporation. All rights reserved.

このソリューション内のプロジェクトを 1 度に 1 つずつビルドします。並行ビルドを有
効にするには、"/m" スイッチを追加してください。
2015/10/30 17:15:08 にビルドを開始しました。
ノード 1 上のプロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C
#\Samples\
WebApp_sample\SPA_Sample\SPA_Sample.sln" (既定のターゲット)。
ValidateSolutionConfiguration:
  ソリューション構成 "Debug|Any CPU" をビルドしています。
プロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\Web
App_s
ample\SPA_Sample\SPA_Sample.sln" (1) は、ノード 1 上に "C:\OpenTouryoTemplates-n
ishino
\root_VS2015\programs\C#\Samples\WebApp_sample\SPA_Sample\SPA_Sample\SPA_Sample
.csproj" (2) をビルドしています (既定のターゲット)。
C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\WebApp_sample\SP
A_Sample\SPA_Sample\SPA_Sample.csproj(320,3): error MSB4019: インポートされたプ
ロジェクト "C:
\Program Files\MSBuild\Microsoft\VisualStudio\v11.0\WebApplications\Microsoft.W
ebApplication.targets" が見つかりませんでした。<Import> 宣言のパスが正しいかどう
か、およびファイルがディスクに存在しているかど
うかを確認してください。
プロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\Web
App_s
ample\SPA_Sample\SPA_Sample\SPA_Sample.csproj" (既定のターゲット) のビルドが終了
しました -- 失敗。

プロジェクト "C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\Web
App_s
ample\SPA_Sample\SPA_Sample.sln" (既定のターゲット) のビルドが終了しました -- 失
敗。


ビルドに失敗しました。

"C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\WebApp_sample\S
PA_Sample\SPA_Sample.sln" (既定のターゲット) (1) ->
"C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\WebApp_sample\S
PA_Sample\SPA_Sample\SPA_Sample.csproj" (既定のターゲット) (2) ->
  C:\OpenTouryoTemplates-nishino\root_VS2015\programs\C#\Samples\WebApp_sample\
SPA_Sample\SPA_Sample\SPA_Sample.csproj(320,3): error MSB4019: インポートされた
プロジェクト "
C:\Program Files\MSBuild\Microsoft\VisualStudio\v11.0\WebApplications\Microsoft
.WebApplication.targets" が見つかりませんでした。<Import> 宣言のパスが正しいかど
うか、およびファイルがディスクに存在している
かどうかを確認してください。

    0 個の警告
    1 エラー

経過時間 00:00:00.80
続行するには何かキーを押してください . . .
```

**Comment by daisukenishino** [comment]
@SandeepNayak077 

The following are problems that have been pointed out from another project.
It is an error that occurred in the VS2013 Express for Web.

"Developer Command Prompt For VS20XX" does not seem to be installed in the express edition.

For this reason it seems to need to consider your own correspondence. Maybe, I think that is needed adding an environment variable in the z_Common.bat.

And in the future, since this change will be merged as "root ---> root_VS2010 ---> root_VS2012 ---> root_VS2013 ---> root_VS2015", You will need validation in each version and edition.

```TXT

C:\root_VS2013\programs\VB>setlocal
--------------------------------------------------
The choice of build configuration (Debug / Release).
--------------------------------------------------
--------------------------------------------------
Creating a build command.
--------------------------------------------------
サブディレクトリまたはファイル Frameworks\Infrastructure\Temp は既に存在します。

サブディレクトリまたはファイル Frameworks\Infrastructure\Build は既に存在します
。
Microsoft (R) Build Engine バージョン 4.6.81.0
[Microsoft .NET Framework、バージョン 4.0.30319.42000]
Copyright (C) Microsoft Corporation. All rights reserved.

このソリューション内のプロジェクトを 1 度に 1 つずつビルドします。並行ビルドを有
効にするには、"/m" スイッチを追加してください。
2015/10/30 15:33:18 にビルドを開始しました。
ノード 1 上のプロジェクト "C:\root_VS2013\programs\VB\Frameworks\Infrastructure\
AllCompone
nt.sln" (既定のターゲット)。
C:\root_VS2013\programs\VB\Frameworks\Infrastructure\AllComponent.sln.metaproj
: error MSB4126: 指定されたソリューション構成 "Debug|BPC" は無効です。構成とプラ
ットフォームのプロパティ (例: MSBuild
.exe Solution.sln /p:Configuration=Debug /p:Platform="Any CPU") を使用して有効な
ソリューション
構成を指定するか、または既定のソリューション構成を使用するために、それらのプロパ
ティを空にしておいてください。 [C:\root_VS2013\program
s\VB\Frameworks\Infrastructure\AllComponent.sln]
プロジェクト "C:\root_VS2013\programs\VB\Frameworks\Infrastructure\AllComponent.
sln"
(既定のターゲット) のビルドが終了しました -- 失敗。


ビルドに失敗しました。

"C:\root_VS2013\programs\VB\Frameworks\Infrastructure\AllComponent.sln" (既定の
ターゲ
ット) (1) ->
(ValidateSolutionConfiguration ターゲット) ->
  C:\root_VS2013\programs\VB\Frameworks\Infrastructure\AllComponent.sln.metapro
j : error MSB4126: 指定されたソリューション構成 "Debug|BPC" は無効です。構成とプ
ラットフォームのプロパティ (例: MSBui
ld.exe Solution.sln /p:Configuration=Debug /p:Platform="Any CPU") を使用して有効
なソリューシ
ョン構成を指定するか、または既定のソリューション構成を使用するために、それらのプ
ロパティを空にしておいてください。 [C:\root_VS2013\progr
ams\VB\Frameworks\Infrastructure\AllComponent.sln]

    0 個の警告
    1 エラー

経過時間 00:00:00.33
Frameworks\Infrastructure\Business\bin\Debug\Business.dll
Frameworks\Infrastructure\Business\bin\Debug\Business.pdb
Frameworks\Infrastructure\Business\bin\Debug\Business.xml
Frameworks\Infrastructure\Business\bin\Debug\Framework.dll
Frameworks\Infrastructure\Business\bin\Debug\Framework.pdb
Frameworks\Infrastructure\Business\bin\Debug\Framework.xml
Frameworks\Infrastructure\Business\bin\Debug\Ionic.Zip.dll
Frameworks\Infrastructure\Business\bin\Debug\log4net.dll
Frameworks\Infrastructure\Business\bin\Debug\Newtonsoft.Json.dll
Frameworks\Infrastructure\Business\bin\Debug\Public.dll
Frameworks\Infrastructure\Business\bin\Debug\Public.pdb
Frameworks\Infrastructure\Business\bin\Debug\Public.xml
Frameworks\Infrastructure\Business\bin\Debug\ja\Business.resources.dll
Frameworks\Infrastructure\Business\bin\Debug\ja\Framework.resources.dll
Frameworks\Infrastructure\Business\bin\Debug\ja\Public.resources.dll
15 個のファイルをコピーしました
Frameworks\Infrastructure\CustomControl\bin\Debug\CustomControl.dll
Frameworks\Infrastructure\CustomControl\bin\Debug\CustomControl.pdb
Frameworks\Infrastructure\CustomControl\bin\Debug\CustomControl.xml
Frameworks\Infrastructure\CustomControl\bin\Debug\Ionic.Zip.dll
Frameworks\Infrastructure\CustomControl\bin\Debug\log4net.dll
Frameworks\Infrastructure\CustomControl\bin\Debug\Newtonsoft.Json.dll
Frameworks\Infrastructure\CustomControl\bin\Debug\Public.dll
Frameworks\Infrastructure\CustomControl\bin\Debug\Public.pdb
Frameworks\Infrastructure\CustomControl\bin\Debug\Public.xml
Frameworks\Infrastructure\CustomControl\bin\Debug\ja\Public.resources.dll
10 個のファイルをコピーしました
Frameworks\Infrastructure\Temp\Debug\Business.dll
Frameworks\Infrastructure\Temp\Debug\Business.pdb
Frameworks\Infrastructure\Temp\Debug\Business.xml
Frameworks\Infrastructure\Temp\Debug\CustomControl.dll
Frameworks\Infrastructure\Temp\Debug\CustomControl.pdb
Frameworks\Infrastructure\Temp\Debug\CustomControl.xml
Frameworks\Infrastructure\Temp\Debug\Framework.dll
Frameworks\Infrastructure\Temp\Debug\Framework.pdb
Frameworks\Infrastructure\Temp\Debug\Framework.xml
Frameworks\Infrastructure\Temp\Debug\Ionic.Zip.dll
Frameworks\Infrastructure\Temp\Debug\log4net.dll
Frameworks\Infrastructure\Temp\Debug\Newtonsoft.Json.dll
Frameworks\Infrastructure\Temp\Debug\Public.dll
Frameworks\Infrastructure\Temp\Debug\Public.pdb
Frameworks\Infrastructure\Temp\Debug\Public.xml
Frameworks\Infrastructure\Temp\Debug\ja\Business.resources.dll
Frameworks\Infrastructure\Temp\Debug\ja\Framework.resources.dll
Frameworks\Infrastructure\Temp\Debug\ja\Public.resources.dll
18 個のファイルをコピーしました
続行するには何かキーを押してください . . .
```

**Comment by daisukenishino** [comment]
@SandeepNayak077 

Except the corresponding Nuget, this correspondence will be performed on [/OpenTouryoProject/OpenTouryo/ repository > develop branch > /root/ folder](https://github.com/OpenTouryoProject/OpenTouryo/blob/develop/root/).

And in the future, since this change will be merged as "root ---> root_easysetup ---> root_VS2010 ---> root_VS2012 ---> root_VS2013 ---> root_VS2015", You will need validation in each version and edition.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

\>> Please investigate the measures of above problem "The build using MSBuild failed".

\>> The following are problems that have been pointed out from another project.
\>> It is an error that occurred in the VS2013 Express for Web.

We have already implemented the batch file.
We have successfully tested the batch file in our environment.
But for confirmation, we want to test the batch file in the environment, that causing the above errors. 

Please give us the eviroment detail to re-produce the above errors, through mail.
We will create the environment or you can share the existing environment with us(Through mail).
If those errors are still exists with the newly modified batch files and Nuget packages, 
we will investigate about issue and report the details of the cause and find the solutions for it.

Please let us know.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

We have found one problem while installing 'IBM.Data.DB2' package from the NuGet.
The installed 'IBM.Data.DB2' v10.0.5.5 package supports only for x64 architecture.
So, in our environment (32-bit OS), when we run '10_Build_WebApp_sample' batch file,
We are getting build error for 'ProjectX_sample' website project. 

Hence, temporarily this package is uninstalled.

Since '\Public\Dll\' folder is deleted, 
When user wants to add 'IBM.Data.DB2' data provoider,
They can install it using 'nuget package manager', with thier own NuGet hosted feeds.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

If the library's Nuget correspondence is not enough, change is not required.
In the above case, changes to DamDB2 project are not required.
In addition, target of Nuget correspondence is root_VS2015 or more.

**Comment by daisukenishino** [comment]
@SandeepNayak077 
> https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-152505385


> https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-152463361
OS  : Windows 8.1 Enterprise 32bit
IDE : Visual Studio Professional 2015

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> If the library's Nuget correspondence is not enough, change is not required.
> In the above case, changes to DamDB2 project are not required.

Shall we uninstall all the packages?
Shall we follow the old procedure as described in the below link?
Or
Shall we remove only the DB2 library?   
For only DB2 library support we can follow the old procedure as described in the below link
Because, other libraries can be updated with latest version using 'nuget package manager' 

Old procedure: https://github.com/OpenTouryoProject/OpenTouryo/tree/develop/root

Please confirm.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>Shall we uninstall all the packages?

Yes, please.

>Shall we follow the [old procedure](https://github.com/OpenTouryoProject/OpenTouryo/tree/develop/root) as described in the below link?

Yes, please.

>Shall we remove only the DB2 library?

What is the difference of "uninstall all the packages" compare with "remove only the DB2 library" ?

>For only DB2 library support we can follow the old procedure as described in the below link
 Because, other libraries can be updated with latest version using 'nuget package manager' 

I understand this.
It should be noted, target of Nuget correspondence is root_VS2015 or more.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

Thank you for the confirmation.
We will uninstall all packages and follow the old procedure.

>What is the difference of "uninstall all the packages" compare with "remove only the DB2 library" ?

"uninstall all the packages": 
Removing all required nuget packages (i.e. Ionic.Zip, log4net, and so on.) 

"remove only the DB2 library":
Removing or uninstalling only DB2 library and install other required nuget packages (i.e. Ionic.Zip, log4net, and so on.)


>It should be noted, target of Nuget correspondence is root_VS2015 or more.

We have targeted Nuget correspondence to only root_VS2015.
We have not targted to any other frameworks i.e. root_2013, root_2012 and so on.
Because if Nuget correspondence to root_VS2015 is successful,
then only we thought to target other frameworks i.e. root_2013, root_2012 and so on.

Please confirm our understanding.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>"uninstall all the packages": 
>Removing all required nuget packages (i.e. Ionic.Zip, log4net, and so on.) 
>
>"remove only the DB2 library":
>Removing or uninstalling only DB2 library and install other required nuget packages (i.e. Ionic.Zip, log4net, and so on.)

I understand this.

>Because if Nuget correspondence to root_VS2015 is successful,
>then only we thought to target other frameworks i.e. root_2013, root_2012 and so on.

No, other frameworks ( i.e. root_2013, root_2012 and so on. ) do not plan to target of Nuget correspondence.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

Thank you for confirmation.
As per the above discussion, we will stop the task of Nuget correspondence and follow the old procedure.
And we will start with investigation of above commented issues https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-152463361 and https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-152467557

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-153545270

Thank you for giving information about the environment.

> OS : Windows 8.1 Enterprise 32bit
> IDE : Visual Studio Professional 2015

This https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-152463361 investigation is under process.

> OS : Windows7 Professional 32bit SP1
> IDE : Visual Studio Express 2013 for Web

We are not able to create this environment.
Because there is no image of Windows7 Professional on Azure VM.
If this environment is already exists on Azure,
Then, is it possible to share the credentials through mail?
So, we can do work around this issue.

Please let us know

**Comment by SandeepNayak077** [comment]
@daisukenishino 

About the issue: https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-152463361
We did research and found one solution.

<b>Problem:</b>

This error occur because of the environment problem that has only VS2015 (i.e. >VS2012).

The MVC and SPA sample projects are built on VS2012.
As per the link http://stackoverflow.com/questions/19718281/external-vs2013-build-error-error-msb4019-the-imported-project-path-was-not
The VS2012 adds the following code part in the <b>csproj</b> file, while creating a new project.
~~~
<PropertyGroup>
    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">10.0</VisualStudioVersion>
    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
</PropertyGroup>
~~~

The above code part were not deleted on upgrade of projects from VS2012 --> VS2013 --> VS2015. 
Because, our environment was installed with VS2012, for this reason we were not getting the this https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-152463361 errors.

<b>Solution:</b>

- Using any text editor, remove the above mentioned code from the "/WebApp_sample/MVC_Sample/MVC_Sample/MVC_Sample.csproj" and "/WebApp_sample/SPA_Sample/SPA_Sample/SPA_Sample.csproj" file.
- After modification again build the projects using MSBuild.

We request you to do the above work around in your environment.
Let us know if the problem is still exists with the above solution. 

If the above solution is OK we will do the code commit with the above solution.
please confirm.

**Comment by daisukenishino** [comment]
@yusukemaegawa 

The above is a report of the solution with the problem of the build error in VS2015.

>We request you to do the above work around in your environment.
>Let us know if the problem is still exists with the above solution. 
>If the above solution is OK we will do the code commit with the above solution.

Please check that there is no problem with the above solution.

**Comment by yusukemaegawa** [comment]
No.
I think that the solution to specify "VisualStudioVersion" parameter is right so that it is written in [the site that you referred to](http://stackoverflow.com/questions/19718281/external-vs2013-build-error-error-msb4019-the-imported-project-path-was-not).
I investigated the "Developer Command Prompt for VS2015", and found that the following environment variable has been set in "Developer Command Prompt for VS2015".
```bat
set VisualStudioVersion=14.0
```
When I set this environment variable, the build error does not occur.
Therefore, I suggest that you set the environment variable mentioned above to z_Common2.bat.

**Comment by daisukenishino** [comment]
@yusukemaegawa 

@SandeepNayak077 

I want to adopt good proposal of above.

**Comment by SandeepNayak077** [comment]
@yusukemaegawa
Thank you for the correction.

@daisukenishino 
\>> I want to adopt good proposal of above.
We will adopt the above proposal.

\>> And, please give priority to the following tasks.
We will modify with the above changes, test it and do code commit by today.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

Is correspondence of this issue completed ?
This change must be included in this merge.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

/>> Is correspondence of this issue completed ?
Yes, it is completed in root_2015.
This changes is already merged using pull request [#21](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/21).

We will include these changes in the merge process.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

I think that it is necessary to complete this fix not only the VS2015 before it is a merge task.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

Do we need to create separate pull request for the above correspondence in the root-->root_easysetup-->root_2010-->root_2012-->root_2013?

please confirm our understanding.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>Do we need to create separate pull request for the above correspondence in the root-->root_easysetup-->root_2010-->root_2012-->root_2013?

I think that the case of the method that such as pullrequest is separated, it is easy for review.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

Yes we understood.
We will create pull request of these changes.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

Regards.
However, this content should be written to #117 instead of #119 (here).

**Comment by daisukenishino** [comment]
@SandeepNayak077 
Please correspond to the following @yusukemaegawa 's pointed out.

- https://github.com/OpenTouryoProject/OpenTouryoTemplates/pull/22/files#r46380458

Note : But, #117 has a higher precedence than #119 (here).

**Comment by SandeepNayak077** [comment]
@daisukenishino 

Sure, we will modify as per the @ 's [pointed out] (https://github.com/OpenTouryoProject/OpenTouryoTemplates/pull/22/files#r46380458). 

> Note : But, #117 has a higher precedence than #119 (here).

We do not hold the task #117 till its completion.

Any intermediate tasks like #119, will be handled in parallel.
We will create new pull request for such tasks with new feature branch.
And we will also take care that it should not affect the merge work.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

I summaried the ["Improvement item" fields](https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issue-111766479).

**Comment by SandeepNayak077** [comment]
@daisukenishino 

>I summaried the "Improvement item" fields.

Thank you for the summary.
We will plan and start this task.

**Comment by daisukenishino** [comment]
### Additional requirements
Finally, I want to remove the packages folder. And, the contents of the packages folder want to be acquired by the NuGet. Therefor, do not remove package.config file from the projects or uninstall nuget packages because when you build the project, the required packages will be downloaded from the NuGet.

But, this changes will affect the direct execution of [this build system](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/10_Build_WebApp_sample.bat#L39). Because, the packages are restored by using the Visual Studio build process. It is not restored by using this [this build system](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/10_Build_WebApp_sample.bat#L39). 

I'm looking for an alternative.

**Comment by yusukemaegawa** [comment]
@daisukenishino 

I think that "[Msbuild Integrated](https://docs.nuget.org/consume/package-restore/msbuild-integrated)" will not affect the current build system.

**Comment by daisukenishino** [comment]
@yusukemaegawa 

>I think that "[Msbuild Integrated](https://docs.nuget.org/consume/package-restore/msbuild-integrated)" will not affect the current build system.

If set to the "Enable NuGet Package Restore" property on Visual Studio project, then it will add .nuget folder containing NuGet.exe and NuGet.targets MSBuild files to solution folder. And, MSBuild will make restore packages folder when perform build task using this settings. Therefor, this will not affect current build system.

I have understood as above, is this correct ?

**Comment by yusukemaegawa** [comment]
@daisukenishino 

Yes, I think so too.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-178390515

As per the investigation, the description in the above comment is correct.

<b>In addition,</b>
The VS2015 has dropped the the feature of "Enable NuGet Package Restore".

We can migrate this feature from VS2013 version.
Or
We must manually configure the [Msbuild-Integrated](https://docs.nuget.org/consume/package-restore/msbuild-integrated) solutions to use Automatic Package Restore.

According to [this link](http://docs.nuget.org/consume/package-restore/migrating-to-automatic-package-restore),
In VS2015, the following files structure must be added to the solution folder
- .nuget (Folder)
 - NuGet.Config
 - NuGet.exe
 - NuGet.targets 

<b>Our question:</b>
For both MVC and SPA sample projects, 
Is it required to add the above file structure and commit to GitHub?
Please let us know.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>We can migrate this feature from VS2013 version.
Or
We must manually configure the Msbuild-Integrated solutions to use Automatic Package Restore.

There is no problem with either method.
However, the subject is only VS2015.

>For both MVC and SPA sample projects, 
Is it required to add the above file structure and commit to GitHub?

We have considered that the followings are target of this correspondance.

- Frameworks
  - AllComponent.sln
    - Business.csproj
    - Framework.csproj
    - Public.csproj
  - Dam
    - DamMySQL.csproj (.sln)
    - DamOraOdp.csproj (.sln)
    - DamPstGrS.csproj (.sln)
    - DamMySQL.csproj (.sln)
    - DamMySQL.csproj (.sln)
  - Tools
     - DeployZipPackWithHTTP.csproj (.sln)
- Samples
  - MVC_Sample.csproj (.sln)
  - SPA_Sample.csproj (.sln)

Please confirm.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-178495691

Oh I see, we thought [this addition requirements](https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-178362320) for only samples project.
Ok, we will do for the above mentioned framworks and samples projects.

The <b>Dam</b> (Data provoiders) projects does not have (.sln) file.

Shall we commit (.sln) file for each Dam projecs?
or
Shall we commit single (.sln) file for all Dam projects, like below?
- /Public/Db/DataProvoiders.sln
 - DamMySQL.csproj 
 - DamOraOdp.csproj 
 - DamPstGrS.csproj 
 - DamDB2 (not having, package.config)
 - DamHiRDB (not having, package.config)

Please let us know.

We request you to add a description about the specification to [here](https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issue-111766479) or [here](https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-178362320).
Because I dont have access to edit your comments.

Currently, the engineers are working with other tasks.
This additional requirement will be taken into next sprint.
Please confirm.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>/Public/Db/DataProvoiders.sln

This is good idea.

>We request you to add a description about the specification to here or here.

I have update [this specification](
https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issue-111766479).

**Comment by SandeepNayak077** [comment]
@daisukenishino 

Thank you for the confirmation.
We have added this task in the plan and assigned to the engineer.
We will modify according to [this specification](https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issue-111766479).

**Comment by daisukenishino** [comment]
@SandeepNayak077 

In review and consultation with @yusukemaegawa,
corresponding method was determined as follows.

- https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/82
  - Almost, there is not a problem.
  - After this has been merged, please send the pullrequest for delete the packages folder.
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/83
  - Set the ["http_proxy" environment variable](https://docs.nuget.org/consume/nuget-config-settings#proxy-settings) in [z_Common.bat](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/z_Common.bat) and [z_Common2.bat](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/z_Common2.bat).
  - Delete the xcopy command of the DLL folder in the [3_Build_Framework.bat](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2015/programs/C%23/3_Build_Framework.bat)

```TXT
xcopy /E /Y "Frameworks\Infrastructure\Public\Dll" "Frameworks\Infrastructure\Build\"
```

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-183152190

Thank you reporting the status of this task.

> https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/82
> After this has been merged, please send the pull request for delete the packages folder.

We did code commit, we are waiting to send next pull request.
Please let us once you merge this pull request.

> https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/83
> Set the "http_proxy" environment variable in z_Common.bat and z_Common2.bat.

This is not possible to test in our environment.
Currently, our organization restricting to set proxy environment.

Please let me know,
What changes needs to be done in z_Common.bat and z_Common2.bat file?
We do the changes according to your suggestion and do code commit.

> https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/83
> Delete the xcopy command of the DLL folder in the 3_Build_Framework.bat

Shall we send this changes in the next pull request?
or
Shall we send this pull request today?
Then, please close this [pull request-83](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/83).

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>What changes needs to be done in z_Common.bat and z_Common2.bat file?
We do the changes according to your suggestion and do code commit.

Meybe like this.
```TXT
@echo --------------------------------------------------
@echo Set the proxy settings of Nuget.
@echo --------------------------------------------------
set http_proxy=XXXXXXXXXX
```

However, please wait.

@yusukemaegawa 

However, I have one question.
Did this download failure really due to proxy settings　?? The problem of part of the package has not been downloaded, I think that there is no relationship to the proxy settings.

**Comment by yusukemaegawa** [comment]
@daisukenishino 

If I did not set "http_proxy" environment variable, the following message showed during build.

"please provide proxy credentials"

And, the following packages were not restored.
(Other packages were restored successfully.)
- jQuery.UI.Combined
- Microsoft.AspNet.Providers.Core
- Microsoft.AspNet.Providers.Core.ja
- Microsoft.AspNet.Providers.LocalDB

If I set "http_proxy" environment variable, the above message did not show and all packages were restored successfully.

However, I do not understand the difference between "the packages that were restored successfully" and "the packages that were not restored".

**Comment by daisukenishino** [comment]
@yusukemaegawa 

I see.
I think that meybe acquisition method of proxy configuration is different for each module.

@SandeepNayak077 

Please appended to the EOF of z_Common.bat and z_Common2.bat as following.

```BAT
@echo --------------------------------------------------
@echo Set the proxy settings of Nuget.
@echo --------------------------------------------------
@rem set http_proxy=http://[username]:[password]@[proxy fqdn or ip address]
```

**Comment by SandeepNayak077** [comment]
@daisukenishino 

Thank you for the confirmation.
We will add the code according to the above comment.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

I plan to merge the following two pull request.
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/82/
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/90/

Result of the merging these pull request,
we recognize that the rest of the work is only to remove the packages folder.

Is this correct ?

If so, After the merge,
Please send a pull request to remove the packages folder.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> Is this correct ?

Along with the removing of packages folder, 
Adding MSBuild integrated to the SPA and MVC samples.
We will send both changes in single pull request with 2 different commits (for easy review).

Please merge the above mentioned pull request if there is no issues.
We will send next pull request.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

Nuget correspondence of [WinAzure_sample.sln](https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015/programs/C%23/Samples/WinAzure_sample) and [WSClientWinStore_sample.sln](https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015/programs/C%23/Samples/WS_sample/WSClient_sample/WSClientWinStore_samples)  is possible ?

(It had been forgotten)

What hours do you need to complete this work ? We expect a rapid response.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-189130564

Yes, these targets were not mentioned earlier.

Yes, It is possible.
We are holding the UI proposal of DaoGen tool task.
And we will send this Nuget correspondence by today EOD.

Is this OK?
Please confirm.
Please

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-189142917
We are holding the UI proposal of DaoGen tool task.
And we will send this Nuget correspondence by today EOD.

I kindly ask for your correspondence.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> And we will send this Nuget correspondence by today EOD.

We did NuGet correspondence for only [WSClientWinStore_sample.sln](https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015/programs/C%23/Samples/WS_sample/WSClient_sample/WSClientWinStore_samples).
Because, we are not able to load [WindowsAzureProject](https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015/programs/C%23/Samples/WinAzure_sample/WindowsAzureProject).

We have downloaded [this](https://www.microsoft.com/en-in/download/details.aspx?id=43709), followed the instruction to instal.
After that we again tried to load [WindowsAzureProject](https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015/programs/C%23/Samples/WinAzure_sample/WindowsAzureProject), but we are not able to load.

Please let us know the environment setup to load the [WindowsAzureProject](https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015/programs/C%23/Samples/WinAzure_sample/WindowsAzureProject) project.

<b>Additional points:</b>
The following projects are present in the [WinAzure_sample.sln](https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2015/programs/C%23/Samples/WinAzure_sample) solution.
- ASPNETWebService
- ProjectX_sample
- WSClientSL_sample

But, we are not planned the above solution projects for [the issue#12](https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12).
Do we also need to plan for these projects?
Please confirm.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>Please let us know the environment setup to load the WindowsAzureProject project.

Is it possible to load the [WinAzure_sample of root_VS2013](https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/master/root_VS2013/programs/C%23/Samples/WinAzure_sample) ?

As far as i searched, Windows Azure SDK for .NET - 2.4 does not support VS2015.

- Download Windows Azure SDK for .NET - 2.4  
from Official Microsoft Download Center  
https://www.microsoft.com/en-in/download/details.aspx?id=43709
    - System Requirements > Required Software:  
Microsoft Visual Studio 2012/Microsoft Visual Studio 2013 OR 
Microsoft Visual Web Developer 2012/Microsoft Visual Web Developer 2013 (only when installing the Windows Azure Tools).

There are Azure SDK for .NET - 2.5 or 2.7
- https://www.microsoft.com/ja-jp/download/details.aspx?id=44938
- https://www.microsoft.com/en-us/download/details.aspx?id=48178

>Additional points:

I have appended the comment.
- see : https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-189128554


>There are more changes required in this project than expected.
Once we can load this project then we can estimate the plan.

I kindly ask for your correspondence.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryo/issues/119#issuecomment-189818750
> Is it possible to load the WinAzure_sample of root_VS2013 ?

We have downloaded and installed Windows Azure SDK for .NET - 2.5.
Now, we can load the project by tageting it to Azure SDK for .NET - 2.5 version.

We did the NuGet correspondence and added portable class library support.
And created new [pull request-100](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/100).
Please merge this pull request and <b>close this issue</b>.

However, the WinAzure_sample.sln project is in an unstable state.
It needs an investigation to resolve it (time consuming).
So, we did not planned this activity in this sprint.
We request you to <b>open new issue</b> to perform the investigation and make the project in a stable state.
We will plan it in the next sprint.
Please let us know.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>And created new pull request-100.
Please merge this pull request and close this issue.

this pull request will be merged after the merge of the following pull request.
https://github.com/OpenTouryoProject/OpenTouryoTemplates/pull/35

>We request you to open new issue to perform the investigation and make the project in a stable state.
We will plan it in the next sprint.

I see. I will do so.
